### ISSUE TITLE
Asking for an outdated key produces an obscure exception

### ISSUE THREAD

**Issue by karol-nowak** [jurmous/etcd4j#92]
A response like this:

```
HTTP/1.1 400 Bad Request
Content-Type: application/json
X-Etcd-Cluster-Id: 7e27652122e8b2ae
X-Etcd-Index: 4429
Date: Tue, 05 Apr 2016 12:29:04 GMT
Content-Length: 154

{"errorCode":401,"message":"The event in requested index is outdated and cleared","cause":"the requested history has been cleared [3430/5]","index":4429}
```

Causes an obscure exception like the one to be thrown:

```
com.fasterxml.jackson.databind.JsonMappingException: Instantiation of [simple type, class mousio.etcd4j.responses.EtcdKeysResponse] value failed: Name is null
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.wrapException(StdValueInstantiator.java:399) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:231) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:135) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:442) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1099) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.module.afterburner.deser.SuperSonicBeanDeserializer.deserializeFromObject(SuperSonicBeanDeserializer.java:236) ~[jackson-module-afterburner-2.7.0.jar:2.7.0]
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:133) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.module.afterburner.deser.SuperSonicBeanDeserializer.deserialize(SuperSonicBeanDeserializer.java:117) ~[jackson-module-afterburner-2.7.0.jar:2.7.0]
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3736) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2796) ~[jackson-databind-2.6.5.jar:2.6.5]
    at mousio.etcd4j.responses.EtcdResponseDecoders$JsonDecoder.decode(EtcdResponseDecoders.java:54) ~[etcd4j-2.10.0.jar:2.10.0]
    at mousio.etcd4j.transport.EtcdResponseHandler.channelRead0(EtcdResponseHandler.java:139) [etcd4j-2.10.0.jar:2.10.0]
    at mousio.etcd4j.transport.EtcdResponseHandler.channelRead0(EtcdResponseHandler.java:47) [etcd4j-2.10.0.jar:2.10.0]
    at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:154) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:439) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:162) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [netty-codec-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:154) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:439) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:162) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:154) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:439) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:162) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:433) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:276) [netty-codec-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:263) [netty-codec-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:243) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:154) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:439) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:162) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:152) [netty-handler-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:154) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:439) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:162) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:924) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:125) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:510) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:467) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:381) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353) [netty-transport-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:742) [netty-common-4.1.0.CR2.jar:4.1.0.CR2]
    at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137) [netty-common-4.1.0.CR2.jar:4.1.0.CR2]
    at java.lang.Thread.run(Thread.java:745) [na:1.8.0_71]
Caused by: java.lang.NullPointerException: Name is null
    at java.lang.Enum.valueOf(Enum.java:236) ~[na:1.8.0_71]
    at mousio.etcd4j.responses.EtcdKeyAction.valueOf(EtcdKeyAction.java:6) ~[etcd4j-2.10.0.jar:2.10.0]
    at mousio.etcd4j.responses.EtcdKeysResponse.<init>(EtcdKeysResponse.java:62) ~[etcd4j-2.10.0.jar:2.10.0]
    at sun.reflect.GeneratedConstructorAccessor42.newInstance(Unknown Source) ~[na:na]
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.8.0_71]
    at java.lang.reflect.Constructor.newInstance(Constructor.java:422) ~[na:1.8.0_71]
    at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call(AnnotatedConstructor.java:125) ~[jackson-databind-2.6.5.jar:2.6.5]
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:227) ~[jackson-databind-2.6.5.jar:2.6.5]
    ... 48 common frames omitted
```

**Comment by lburgazzoli** [comment]
What is the etcd4j version?

**Comment by karol-nowak** [comment]
Sorry, forgot to mention that: 2.10.0

**Comment by lburgazzoli** [comment]
I think this has been solved by https://github.com/jurmous/etcd4j/issues/80, could you re-test with the latest code ?

**Comment by lburgazzoli** [comment]
@jurmous would it be possible to release a new version ?

**Comment by jurmous** [comment]
I did a snapshot 2.10.2-SNAPSHOT today. I think it is best to wait for full release with latest issues and activity on that.

**Comment by jurmous** [comment]
Released 2.11.0
